lms/djangoapps/instructor/tests/test_api.py

```diff
@@ -2022,6 +2033,190 @@ def test_get_students_features_cohorted(self, is_cohorted):
 
         self.assertEqual('cohort' in res_json['feature_names'], is_cohorted)
 
+    def test_enrollment_report_features_csv(self):
+        """
+        test to generate enrollment report.
+        enroll users, admin staff using registration codes.
+        """
+        invoice_transaction = InvoiceTransaction(
+            invoice=self.sale_invoice_1,
+            amount=self.sale_invoice_1.total_amount,
+            status='completed',
+            created_by=self.instructor,
+            last_modified_by=self.instructor
+        )
+        invoice_transaction.save()
+        course_registration_code = CourseRegistrationCode(
+            code='abcde',
+            course_id=self.course.id.to_deprecated_string(),
+            created_by=self.instructor,
+            invoice=self.sale_invoice_1,
+            invoice_item=self.invoice_item,
+            mode_slug='honor'
+        )
+        course_registration_code.save()
+
+        admin_user = AdminFactory()
+        admin_cart = Order.get_cart_for_user(admin_user)
+        PaidCourseRegistration.add_to_order(admin_cart, self.course.id)
+        admin_cart.purchase()
+
+        test_user = UserFactory()
+        redeem_url = reverse('register_code_redemption', args=[course_registration_code.code])
+        self.client.login(username=test_user.username, password='test')
+        response = self.client.get(redeem_url)
+        self.assertEquals(response.status_code, 200)
+        # check button text
+        self.assertTrue('Activate Course Enrollment' in response.content)
+
+        response = self.client.post(redeem_url)
+        self.assertEquals(response.status_code, 200)
+
+        CourseFinanceAdminRole(self.course.id).add_users(self.instructor)
+        user_profile_1 = UserProfileFactory.create(user=self.students[0])
+        user_profile_1.meta = '{"company": "asdasda"}'
+        user_profile_1.save()
+
+        # UserFactory.profile(self.students[0], create=True, extracted=None, {"meta": {"company": 'Test com'}})
+        self.client.login(username=self.instructor.username, password='test')
+        url = reverse('get_enrollment_report', kwargs={'course_id': self.course.id.to_deprecated_string()})
+        response = self.client.get(url, {})
+        self.assertIn('Your detailed enrollment report is being generated!', response.content)
+
+    def test_bulk_purchase_detailed_report(self):
+        """
+        test to generate detailed enrollment report.
```

Comment:
This would be a bit clearer as a numbered list of steps this tests is doing:

1. Purchase registration codes.
2. Enroll users via registration code.
3. Validate generated enrollment report.